bulls_and_cows/lib/bulls_and_cows.ex

```diff
@@ -0,0 +1,158 @@
+defmodule BullsAndCows do
+  use Accessible
+
+  @moduledoc """
+  A game of bulls and cows, rules:
+  You need to guess 4 different numbers in definite order.
+  If number is correct but in wrong position, its a cow, if value and position is correct, its a bull.
+
+  Game should be played in iex. First you init game by new_game/0,
+  then you should pass state to make_turn/2 until you win
+
+  Game example:
+  $ game = BullsAndCows.new_game()
+  %BullsAndCows{number_encrypted: "ೞ", seed: 314, moves: [], win: false}
+  """
+
+  @seed_range 500
+
+  defstruct number_encrypted: [],
+            seed: 0,
+            moves: [],
+            win: false
+
+  @doc """
+  Returns game state
+  """
+  def new_game() do
+    number = generate_number_list()
+    seed = Enum.random(0..@seed_range)
+
+    %__MODULE__{
+      number_encrypted: encrypt(number),
+      seed: seed,
+      moves: [],
+      win: false
+    }
+  end
+
+  @doc """
+  Updates state with your answer, checks win conditions
+
+  ## Examples
+
+      iex> BullsAndCows.make_turn(%BullsAndCows{number_encrypted: "ೞ", seed: 314, moves: [], win: false}, 1234)
+      %BullsAndCows{
+        number_encrypted: "ೞ",
+        seed: 314,
+        moves: [%{answer: 1234, bulls: 0, cows: 1}],
+        win: false
+      }
+
+      iex> BullsAndCows.make_turn(%BullsAndCows{number_encrypted: "ڊ", seed: 411, moves: [%{answer: 1263, bulls: 4, cows: 0}], win: true}, 8888)
+      %{win: true, moves_cnt: 1}
+
+  """
+  def make_turn(%__MODULE__{win: true, moves: moves}, _turn),
```

Comment:
What’s the actual meaning of calling `make_turn/2` when the game is already won? I’d return an error, saying “you already won,” or simply omit this clause making the code raise.